vllm/v1/worker/gpu_model_runner.py

```diff
@@ -1288,9 +1288,18 @@ def _dummy_sampler_run(
             allowed_token_ids_mask=None,
             bad_words_token_ids={},
         )
-        sampler_output = self.model.sample(logits=logits,
-                                           sampling_metadata=dummy_metadata)
-
+        try:
+            sampler_output = self.model.sample(
+                logits=logits, sampling_metadata=dummy_metadata)
+        except RuntimeError as e:
+            if 'out of memory' in str(e):
+                raise RuntimeError(
+                    "CUDA out of memory occurred when warming up sampler with "
+                    f"{num_reqs} dummy requests. Please try lowering "
+                    "`max_num_seqs` or `gpu_memory_utilization` when "
+                    "initializing the engine.") from None
```

Comment:
```suggestion
                    "initializing the engine.") from e
```

Do you want to hide the stacktrace or inherit it?

Reply:
I think it'll be too noisy/confusing if we show the entire trace and this check is already trying to catch `out of memory` errors, but happy to change anyways.

Reply:
also vote for "e" :-)

Reply:
That's fine - updated!